rfc/rfc-46/rfc-46.md

```diff
@@ -156,13 +187,76 @@ Following major components will be refactored:
 3. `HoodieRealtimeRecordReader`s 
    1. API will be returning opaque `HoodieRecord` instead of raw Avro payload
 
+### Config for Record Merge
+The MERGE_CLASS_NAME config is engine-aware. If you are not specified the MERGE_CLASS_NAME, MERGE_CLASS_NAME will be specified default according to your engine type.
+
+### Public Api in HoodieRecord
+Because we implement different types of records, we need to transfer some func in AvroUtils into HoodieRecord for different data(avro, InternalRow, RowData).
+Its public API will look like following:
+
+```java
+class HoodieRecord {
+
+   /**
+    * Get column in record to support RDDCustomColumnsSortPartitioner
+    */
+   Object getRecordColumnValues(Schema recordSchema, String[] columns, boolean consistentLogicalTimestampEnabled);
```

Comment:
I'd suggest we go with singular option: 
```
<T> T getField(String column)
```

We should not be passing the schema in, schema should be kept inside HoodieRecord itself. Also, let's get rid of `consistentLogicalTimestampEnabled` you can just pass it as true in your implementation.

Reply:
HoodieAvroRecord(HoodieRecordPayload) should not hold the schema. This will bring extra memory overhead for each record. So the recordSchema is used by HoodieRecordPayload.